cookbook/workflows/README.md

```diff
@@ -0,0 +1,499 @@
+# Agno Workflows 2.0 - Developer Guide
+
+Welcome to **Agno Workflows 2.0** - the next generation of intelligent, flexible workflow orchestration. This guide covers all workflow patterns, from simple linear sequences to complex conditional logic with parallel execution.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Core Concepts](#core-concepts)
+- [Workflow Patterns](#workflow-patterns)
+  - [1. Sequential Workflows](#1-sequential-workflows)
+  - [2. Parallel Execution](#2-parallel-execution)
+  - [3. Conditional Workflows](#3-conditional-workflows)
+  - [4. Loop/Iteration Workflows](#4-loopiteration-workflows)
+  - [5. Router-Based Branching](#5-router-based-branching)
+  - [6. Mixed Execution Types](#6-mixed-execution-types)
+  - [7. Function-Based Steps](#7-function-based-steps)
+  - [8. Complex Combinations](#8-complex-combinations)
+- [Advanced Features](#advanced-features)
+- [Best Practices](#best-practices)
+- [Migration from Workflows 1.0](#migration-from-workflows-10)
+
+## Overview
+
+Agno Workflows 2.0 provides a powerful, declarative way to orchestrate multi-step AI processes. Unlike traditional linear workflows, you can now create sophisticated branching logic, parallel execution, and dynamic routing based on content analysis.
+
+### Key Features
+
+- 🔄 **Flexible Execution**: Sequential, parallel, conditional, and loop-based execution
+- 🎯 **Smart Routing**: Dynamic step selection based on content analysis
+- 🔧 **Mixed Components**: Combine agents, teams, and functions seamlessly
+- 💾 **State Management**: Share data across steps with session state
+- 🌊 **Streaming Support**: Real-time processing with streaming capabilities
+- 📝 **Structured Inputs**: Type-safe inputs with Pydantic models
+
+## Core Concepts
+
+### Building Blocks
+
+| Component | Purpose | Example Use Case |
+|-----------|---------|------------------|
+| **Step** | Basic execution unit | Single research task |
+| **Agent** | AI assistant with specific role | Content writer, researcher |
+| **Team** | Coordinated group of agents | Research team with specialists |
+| **Function** | Custom Python logic | Data processing, API calls |
+| **Parallel** | Concurrent execution | Multiple research streams |
+| **Condition** | Conditional execution | Topic-specific processing |
+| **Loop** | Iterative execution | Quality-driven research |
+| **Router** | Dynamic routing | Content-based step selection |
+
+
+## Workflow Patterns
+
+### 1. Sequential Workflows
+
+**When to use**: Linear processes where each step depends on the previous one.
+
+**Example**: Research → Analysis → Content Creation
+
+```python
+from agno.workflow.v2 import Step, Workflow
+
+# Simple sequential workflow
+workflow = Workflow(
+    name="Content Creation Pipeline",
+    steps=[
```

Comment:
I would use an even simpler one where you don't have to use `Step`, just use a list of agents.

Then say "You can name your steps" for better logging, and future support on the Agno platform, something like that.